### PR TITLE
[`Print Search Comment`] new tweak

### PR DESCRIPTION
**Pull request by Infiziert90**
A simple tweak that prints the Search Comment into chat.
Only triggers if the user inspects them in person, for example via `/c <mo>` macro or using the context menu option "Examine"

Example (name censored by chat2s screenshot mode)
image

### CODE REVIEW

**Comment by Infiziert90** [comment]
Updated for api X

**Comment by Infiziert90** [comment]
Fixed for the new SimpleTweaks version
Set the version to 0.0.0.0 as unreleased was dropped

**Comment by Caraxi** [comment]
Unreleased should exist still, just only in debug builds on the plugin

**Comment by Infiziert90** [comment]
True, must have put it on release at some point without noticing

**Comment by Infiziert90** [comment]
Switched to Lumina for SeString builder, dalamud's version of PlayerPayload is bugged and crashes with newer worlds (encoding WorldId as byte instead of ushort)

**Comment by Caraxi** [comment]
oops didn't mean to close that

File: Tweaks/Chat/PrintSearchComment.cs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+[TweakReleaseVersion(UnreleasedVersion)]
+public class PrintSearchComment : ChatTweaks.SubTweak {
+    [TweakHook]
+    private HookWrapper<AgentInspect.Delegates.ReceiveSearchComment>? receiveSearchComment;
+
+    protected override unsafe void Setup()
+    {
+        base.Setup();
+
+        receiveSearchComment ??= Common.Hook<AgentInspect.Delegates.ReceiveSearchComment>(AgentInspect.MemberFunctionPointers.ReceiveSearchComment, ReceiveSearchCommentDetour);
```
Comment (Caraxi) [inline Tweaks/Chat/PrintSearchComment.cs:24]: Setup is run even if the tweak is not enabled, so should not be enabling hooks here

**Pull request merged**